examples/foo-eds_v4_image_pull_error.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: datadoghq.com/v1alpha1
```

Comment:
I think we should not put e2e tests manifest in another than the `examples` folder. maybe `controllers/test-data`

Reply:
What about if we add a `README.md` under examples - that way we can explain what each example intends to demonstrate. The e2e tests don't depend on these new files per se, my thought was to demonstrate certain features and behaviors per se, but I agree that it might be confusing to others.

Or maybe you meant that these should be only showing settings and in the happy case only. These indeed will cause canary to handle in abnormal scenarios.

Reply:
Yes I mean `examples` should only contains working examples.
But it could be also interesting for users, to see how the canary works. So maybe you can add a subfolder `examples/canary-tests`. WDYT ?